cmake/kokkos_enable_options.cmake

```diff
@@ -58,6 +58,7 @@ KOKKOS_ENABLE_OPTION(IMPL_CUDA_UNIFIED_MEMORY OFF "Whether to leverage unified m
 KOKKOS_ENABLE_OPTION(DEPRECATED_CODE_4    ON "Whether code deprecated in major release 4 is available" )
 KOKKOS_ENABLE_OPTION(DEPRECATION_WARNINGS ON "Whether to emit deprecation warnings" )
 KOKKOS_ENABLE_OPTION(HIP_RELOCATABLE_DEVICE_CODE  OFF "Whether to enable relocatable device code (RDC) for HIP")
+KOKKOS_ENABLE_OPTION(IMPL_SYCL_RELOCATABLE_DEVICE_CODE  ON "Whether to enable relocatable device code (RDC) for SYCL")
```

Comment:
Mark it as advanced?

Reply:
I think we should turn it off by default.

Reply:
Also should that default be `OFF` instead?

Reply:
Please elaborate why you think the right choice is `ON` by default.
At first sight it clashes with what we do in the HIP and CUDA backends.